### PR TITLE
Set up unit testing in dotnet

### LINKED ISSUE #172: Set up unit testing in dotnet

### PR DESCRIPTION
**Pull request by GodVenn**
Closes #172

### CODE REVIEW

File: backend/api.test/TestRobotController.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        [InlineData(5, typeof(NotFoundResult))]
+        public void GetRobotById(int robotId, Type actionResultType)
+        {
+            IActionResult? result = _controller.GetRobotById(robotId.ToString()).Result.Result;
+
+            // Check if is type or inherited from type
+            Assert.IsAssignableFrom(actionResultType, result);
+        }
+    }
+}
```
Comment (Christdej) [inline backend/api.test/TestRobotController.cs:38]: Add newline please

File: backend/api/Models/Models.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@ namespace api.Models
     public class Robot
     {
         [Key]
-        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        //[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
```
Comment (Christdej) [inline backend/api/Models/Models.cs:10]: Ideally I would not comment out this line and make the test work with autogenerated IDs. Also remove the IDs from the InitDb.cs file, as we always want the DB to create the IDs

File: backend/api/Context/InitDb.cs [hunk 3]
```diff
@@ -18,6 +19,7 @@ private static List<Robot> GetRobots()
 
         Robot robot2 = new Robot
         {
+            Id = "2",
```
Comment (Christdej) [inline backend/api/Context/InitDb.cs:22]: Remove, see comment below

File: backend/api/Context/InitDb.cs [hunk 4]
```diff
@@ -10,6 +10,7 @@ private static List<Robot> GetRobots()
     {
         Robot robot1 = new Robot
         {
+            Id = "1",
```
Comment (Christdej) [inline backend/api/Context/InitDb.cs:13]: Remove, see comment below

File: backend/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# Flotilla backend
+
+## Setup
+
+TODO
+
+## Install
+
+TODO
+
```
Comment (Christdej) [inline backend/README.md:10]: Remove these lines for now

**Review by Christdej — CHANGES_REQUESTED** [review]

File: backend/api/Models/Models.cs (hunk 2 above)
Comment (GodVenn) [reply backend/api/Models/Models.cs]: Fixed

File: backend/api/Context/InitDb.cs (hunk 3 above)
Comment (GodVenn) [reply backend/api/Context/InitDb.cs]: Fixed

File: backend/README.md (hunk 5 above)
Comment (GodVenn) [reply backend/README.md]: Fixed

File: backend/api/Context/InitDb.cs (hunk 4 above)
Comment (GodVenn) [reply backend/api/Context/InitDb.cs]: Fixed

**Review by Christdej — APPROVED** [review]
LGTM

**Pull request merged**